js/packages/teams-ai/src/Application.ts

```diff
@@ -747,6 +747,114 @@ export class Application<TState extends TurnState = DefaultTurnState> {
     }
 }
 
+/**
+ * A builder class for simplifying the creation of an Application instance.
+ * @template TState Optional. Type of the turn state. This allows for strongly typed access to the turn state.
+ */
+export class ApplicationBuilder<TState extends TurnState = DefaultTurnState> {
+    private _options: Partial<ApplicationOptions<TState>> = {};
+
+    /**
+     * Configures the application to use long running messages.
+     * Default state for longRunningMessages is false
+     * @param {BotAdapter} adapter The adapter to use for routing incoming requests.
+     * @param {string} botAppId The Microsoft App ID for the bot.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withLongRunningMessages(adapter: BotAdapter, botAppId: string): this {
+        if (!botAppId) {
+            throw new Error(
+                `The Application.longRunningMessages property is unavailable because botAppId cannot be null or undefined.`
+            );
+        }
+
+        this._options.longRunningMessages = true;
+        this._options.adapter = adapter;
+        this._options.botAppId = botAppId;
+        return this;
+    }
+
+    /**
+     * Configures the storage system to use for storing the bot's state.
+     * @param {Storage} storage The storage system to use.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withStorage(storage: Storage): this {
+        this._options.storage = storage;
+        return this;
+    }
+
+    /**
+     * Configures the AI system to use for processing incoming messages.
+     * @param {AIOptions<TState>} aiOptions The options for the AI system.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withAI(aiOptions: AIOptions<TState>): this {
+        this._options.ai = aiOptions;
+        return this;
+    }
+
+    /**
+     * Configures the turn state manager to use for managing the bot's turn state.
+     * @param {TurnStateManager<TState>} turnStateManager The turn state manager to use.
+     * @returns {this} The ApplicationBuilder instance.
+     */
+    public withTurnStateManager(turnStateManager: TurnStateManager<TState>): this {
```

Comment:
Wasn't this going to be renamed to ~Options?

Reply:
Same with AI and Storage above

Reply:
My understanding was appending 'options' for adaptive cards and task modules but happy to do that for AI as well. I think we should keep storage and turn state manager as is however, considering an object is being passed in vs specific options.

Reply:
@aacebo @singhk97 what are your thoughts? I can see how Turn State Manager and Storage are unique since they have specific objects passed in.

Reply:
I think it should be named according to whats being passed in, so this this isn't passing in an options interface but instead the instance itself the naming feels accurate.

Reply:
Okay, sounds good to me. In the case of AI, I would say it's passing in options -- does that sound okay?